plugins/restful/RestfulFilesUpload.php

```diff
@@ -26,6 +26,8 @@ public static function controllersInfo() {
    * - "validators": By default no validation is done on the file extensions or
    *   file size.
    * - "scheme": By default the default scheme (e.g. public, private) is used.
+   * - "make_permanent": When true the file's status will be set to permanent
```

Comment:
A static value like that would suffice for a just creating the reference to prevent deletion attempts. 

The only place core displays those values is when logging the failed deletion: https://github.com/drupal/drupal/blob/7.x/modules/system/system.module#L3047
But I think that at least shows the intention of the values.

If you look at the some of the places in core where they're set you'll see all three parameters (module, type, id) are used to indicate which module is using the file and how it's using it:
https://github.com/drupal/drupal/blob/7.x/modules/file/file.field.inc#L245 :
```php
file_usage_add($file, 'file', $entity_type, $id);
```
https://github.com/drupal/drupal/blob/7.x/modules/image/image.module#L458 :
```php
file_usage_add($file_new, 'image', 'default_image', $field['id']);
```
https://github.com/drupal/drupal/blob/7.x/modules/user/user.module#L491 :
```php
file_usage_add($picture, 'user', 'user', $account->uid);
```
The idea there being that you could build links to the usages.

So I guess my point is really that the caller should probably be responsible for registering a usage. If you're putting those files into image or file fields, they'll be registered there automatically. If you're storing it elsewhere you should manage your own usage so the files can be deleted. 

That could be a big change in behavior so perhaps the change should only happen in the 2.x branch. The `set_usage` option could be there just for those who want to keep the 1.x behavior.